src/apps/loan-list/materials/utils/material-fetch-hoc.tsx

```diff
@@ -1,82 +1,68 @@
-import React, { useEffect, useState, ComponentType } from "react";
-import {
-  GetMaterialManifestationQuery,
-  useGetMaterialManifestationQuery
-} from "../../../../core/dbc-gateway/generated/graphql";
-import { LoanMetaDataType } from "../../../../core/utils/types/loan-meta-data-type";
+import React, { useEffect, useState, ComponentType, FC } from "react";
+import { useGetMaterialManifestationQuery } from "../../../../core/dbc-gateway/generated/graphql";
+import { Product } from "../../../../core/publizon/model";
 import { FaustId } from "../../../../core/utils/types/ids";
+import { BasicDetailsType } from "../../../../core/utils/types/basic-details-type";
+import { mapManifestationToBasicDetailsType } from "../../../../core/utils/helpers/list-mapper";
 
-export interface MaterialDetailsProps {
-  loanMetaData: LoanMetaDataType;
+export interface MaterialProps {
+  material?: BasicDetailsType | null;
 }
 
-export interface SelectableMaterialProps {
-  loanMetaData: LoanMetaDataType;
-  disabled?: boolean;
-  materialsToRenew?: number[];
-  onChecked?: (faust: FaustId) => void;
-}
+type InputProps = {
+  digitalMaterial?: Product | null;
+  faust: FaustId | null;
+  identifier: string | null;
+};
 
-export interface StackableMaterialProps {
-  stack?: LoanMetaDataType[];
-  loanMetaData: LoanMetaDataType;
-  amountOfMaterialsWithDueDate?: number;
-  dueDateLabel?: string;
-  openModal?: boolean;
-  selectMaterial?: ({
-    material,
-    loanMetaData
-  }: {
-    material: GetMaterialManifestationQuery | undefined | null;
-    loanMetaData: LoanMetaDataType;
-  }) => void;
-}
+const fetchMaterial =
+  <P extends object>(
+    Component: ComponentType<P & MaterialProps>
+  ): FC<P & InputProps> =>
+  ({ identifier, faust, ...props }: InputProps) => {
+    // If this is a digital book, another HOC fetches the data and this
+    // HOC just returns the component
+    if (identifier) {
+      return (
+        <Component
+          /* eslint-disable-next-line react/jsx-props-no-spreading */
+          {...(props as P)}
+          identifier={identifier}
+        />
+      );
+    }
 
-export interface MaterialProps {
-  material: GetMaterialManifestationQuery;
-}
+    if (faust) {
+      const [material, setMaterial] = useState<BasicDetailsType>();
 
-type InputProps = { loanMetaData: LoanMetaDataType } & (
-  | StackableMaterialProps
-  | SelectableMaterialProps
-  | MaterialDetailsProps
-);
-export type WithMaterialProps = MaterialProps &
-  (SelectableMaterialProps | StackableMaterialProps | MaterialDetailsProps);
+      // Todo error handling
+      const { isSuccess: isSuccessManifestation, data } =
+        useGetMaterialManifestationQuery({
```

Comment:
This hook name is too generic. We could easily have other cases where we would need to fetch a Material Manifestation but with other fields. I think it should be renamed by changing the query name.

Reply:
I am adding some fields to it in a later pr, so perhaps I can rename it in that?

Reply:
👍